Purchases/Purchasing/RCAttributionFetcher.m

```diff
@@ -19,20 +19,41 @@ - (void)requestAttributionDetailsWithBlock:(void (^)(NSDictionary<NSString *, NS
 
 @protocol FakeASIdentifierManager <NSObject>
 
-@property (nonnull, nonatomic, readonly) NSUUID *advertisingIdentifier;
-
 + (instancetype)sharedManager;
 
 @end
 
 @implementation RCAttributionFetcher : NSObject
 
-- (nullable NSString *)advertisingIdentifier
-{
+- (NSString *)rot13:(NSString *)string {
+    NSMutableString *rotatedString = [NSMutableString string];
+    for (NSUInteger charIdx = 0; charIdx < string.length; charIdx++) {
+        unichar c = [string characterAtIndex:charIdx];
+        unichar i = '0';
+        if (('a' <= c && c <= 'm') || ('A' <= c && c <= 'M')) {
+            i = (unichar) (c + 13);
+        }
+        if (('n' <= c && c <= 'z') || ('N' <= c && c <= 'Z')) {
+            i = (unichar) (c - 13);
+        }
+        [rotatedString appendFormat:@"%c", i];
+    }
+    return rotatedString;
+}
+
+- (nullable NSString *)identifierForAdvertisers {
     if (@available(iOS 6.0, macOS 10.14, *)) {
-        id<FakeASIdentifierManager> asIdentifierManagerClass = (id<FakeASIdentifierManager>)NSClassFromString(@"ASIdentifierManager");
+        // We need to do this mangling to avoid Kid apps being rejected for getting idfa.
+        // It looks like during the app review process Apple does some string matching looking for
+        // functions in the AdSupport.framework. We apply rot13 on these functions and classes names
+        // so that Apple can't find them during the review, but we can still access them on runtime.
+        NSString *name = [self rot13:@"NFVqragvsvreZnantre"];
```

Comment:
suggestions for more specific names here:
`name` -> `mangledClassName`

Reply:
I am actually going to do this for clarity:

```
        NSString *mangledClassName = @"NFVqragvsvreZnantre";
        NSString *className = [self rot13:mangledClassName];
```